### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by superpooperxxx**
[DEMO](https://superpooperxxx.github.io/react_dynamic-list-of-todos/)

### CODE REVIEW

**Review by Fynkzia — APPROVED** [review]
Great🔥

File: src/App.tsx [hunk 1]
```diff
@@ -1,14 +1,46 @@
...
+  const [statusOption, setStatusOption] = useState('all');
+  const [query, setQuery] = useState('');
+
+  useEffect(() => {
+    try {
+      getTodos()
+        .then(setTodos);
+    } catch (error) {
+      setTodos([]);
+    }
```
Comment (Ihor-Karpyn) [inline src/App.tsx:31]: if u use `.then` use `.catch` method instead of try/catch construction

File: src/App.tsx [hunk 2]
```diff
@@ -1,14 +1,46 @@
...
+
+  const selectedTodo = useMemo(() => (
+    todos.find(todo => todo.id === selectedTodoId)
+  ), [selectedTodoId]);
+
+  const handleCloseModal = useCallback(() => {
+    setSelectedTodoId(0);
+  }, []);
+
+  const visibleTodos = getVisibleTodos(todos, query, statusOption);
```
Comment (Ihor-Karpyn) [inline src/App.tsx:42]: 🔥

**Review by Ihor-Karpyn — APPROVED** [review]
Good job

**Pull request closed without merging**